cpmpy/expressions/globalconstraints.py

```diff
@@ -291,6 +299,52 @@ def value(self):
         rev = [argval(a) for a in self.args[1]]
         return all(rev[x] == i for i, x in enumerate(fwd))
 
+class InverseOne(GlobalConstraint):
+    """
+       Inverse (aka channeling / assignment) constraint but with only one array.
+       Equivalent to Inverse(x,x)
+
+                arr[i] == j  <==>  arr[j] == i
+
+    """
+    def __init__(self, arr):
+        flatargs = flatlist([arr])
+        if any(is_boolexpr(arg) for arg in flatargs):
+            raise TypeError("Only integer arguments allowed for global constraint Inverse: {}".format(flatargs))
+        super().__init__("inverseOne", [arr])
+
+    def decompose(self):
+        from .python_builtins import all
+        arr = self.args[0]
+        arr = cpm_array(arr)
+        return [all(arr[x] == i for i, x in enumerate(arr))], []
+
+    def value(self):
+        valsx = [argval(a) for a in self.args[0]]
+        return all(valsx[x] == i for i, x in enumerate(valsx))
+
+class Channel(GlobalConstraint):
+    """
+        Channeling constraint. Channeling integer representation of a variable into a representation with boolean
```

Comment:
Nice. Which constraint from the global constraint catalog does it correspond to?
Is there a more appropriate name? e.g. ChannelToBool or something, because it is not clear right now that it concerns int to bool represenation, and was confused seeing the name what is the difference with inverse

Reply:
In XCSP3, all of these correspond to Channel... 
I can't seem to find this one in the global constraint catalog (closest is link_set_to_boolean but do not do that exactly)
The InverseOne is Inverse in the catalog.. The currently named inverse (symmetric) is the inverse_set in the global constraint catalog. Inverse asymetric is nowhere in the catalog.